src/CMakeLists.txt

```diff
@@ -226,69 +240,91 @@ register_migraphx_ops(
 )
 register_op(migraphx HEADER migraphx/op/rnn_variable_seq_lens.hpp OPERATORS op::rnn_var_sl_shift_output op::rnn_var_sl_shift_sequence)
 register_op(migraphx HEADER migraphx/builtin.hpp OPERATORS builtin::literal builtin::param builtin::returns)
-rocm_clang_tidy_check(migraphx)
+
+if(CLANG_TIDY_EXE)
+    rocm_clang_tidy_check(migraphx)
+endif()
+
 rocm_install_targets(
   TARGETS migraphx
   INCLUDE
     ${CMAKE_CURRENT_SOURCE_DIR}/include
     ${CMAKE_CURRENT_BINARY_DIR}/include
 )
 
-
-check_cxx_linker_flag(-lstdc++fs HAS_LIB_STD_FILESYSTEM)
-if(HAS_LIB_STD_FILESYSTEM)
-target_link_libraries(migraphx PRIVATE -lstdc++fs)
+if(NOT WIN32)
+    check_cxx_linker_flag(-lstdc++fs HAS_LIB_STD_FILESYSTEM)
+    if(HAS_LIB_STD_FILESYSTEM)
+        target_link_libraries(migraphx PRIVATE -lstdc++fs)
+    endif()
+    target_link_libraries(migraphx PRIVATE -ldl)
 endif()
 
-target_link_libraries(migraphx PRIVATE -ldl)
-
 target_include_directories(migraphx SYSTEM PUBLIC $<BUILD_INTERFACE:${HALF_INCLUDE_DIR}>)
-
-find_package(Threads)
 target_link_libraries(migraphx PUBLIC Threads::Threads)
 
-find_package(nlohmann_json 3.8.0 REQUIRED)
+if(NOT WIN32)
+    find_package(nlohmann_json 3.8.0 REQUIRED)
+endif()
+
 target_link_libraries(migraphx PRIVATE nlohmann_json::nlohmann_json)
 
-find_package(PkgConfig)
-pkg_check_modules(SQLITE3 REQUIRED IMPORTED_TARGET sqlite3)
-target_link_libraries(migraphx PRIVATE PkgConfig::SQLITE3)
+generate_export_header(migraphx
+    EXPORT_FILE_NAME
+        ${CMAKE_BINARY_DIR}/include/migraphx/export.h)
+
+target_include_directories(migraphx PRIVATE ${CMAKE_BINARY_DIR}/include)
```

Comment:
So this needs to be `PUBLIC` since this is included in the headers that downstream consumers will use(such as the test). This is a reason why it would be best to have a `export_dll` function to encapsulate all these fine details.

Reply:
Fixed on `uif2-initial` branch - not yet pushed to repo.